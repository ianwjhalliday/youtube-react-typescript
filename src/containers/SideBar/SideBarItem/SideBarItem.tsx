import React from 'react';
import cx from 'classnames';
import {Icon, Menu, SemanticICONS} from 'semantic-ui-react';
import './SideBarItem.scss';

export interface SideBarItemProps {
  highlight?: boolean;
  icon: SemanticICONS;
  label: string;
}

export const SideBarItem: React.FC<SideBarItemProps> = (props) => {
  const {highlight, icon, label} = props;
  const classNames = cx({
    'sidebar-item': true,
    'highlight-item': highlight,
  });
  return (
    <Menu.Item className={classNames}>
      <div className="sidebar-item-alignment-container">
        <span><Icon size="large" name={icon}/></span>
        <span>{label}</span>
      </div>
    </Menu.Item>
  )
};
