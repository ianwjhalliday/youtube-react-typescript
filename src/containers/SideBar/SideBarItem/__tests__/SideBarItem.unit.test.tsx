import React from 'react';
import {shallow} from 'enzyme';
import SideBarItem from '../SideBarItem';

describe('SideBarItem', () => {
  describe('renders SideBarItem with a label and icon', () => {
    test('without highlight prop', () => {
      const wrapper = shallow(
        <SideBarItem label="a label" icon="bell"/>
      );
      expect(wrapper).toMatchSnapshot();
    });

    test('highlight prop is false', () => {
      const wrapper = shallow(
        <SideBarItem label="a label" icon="bell" highlight={false}/>
      );
      expect(wrapper).toMatchSnapshot();
    });

    test('highlight prop is true', () => {
      const wrapper = shallow(
        <SideBarItem label="a label" icon="bell" highlight/>
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
