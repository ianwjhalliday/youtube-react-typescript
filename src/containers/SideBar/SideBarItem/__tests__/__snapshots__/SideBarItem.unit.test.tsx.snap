// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBarItem renders SideBarItem with a label and icon highlight prop is false 1`] = `
<MenuItem
  className="sidebar-item"
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        name="bell"
        size="large"
      />
    </span>
    <span>
      a label
    </span>
  </div>
</MenuItem>
`;

exports[`SideBarItem renders SideBarItem with a label and icon highlight prop is true 1`] = `
<MenuItem
  className="sidebar-item highlight-item"
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        name="bell"
        size="large"
      />
    </span>
    <span>
      a label
    </span>
  </div>
</MenuItem>
`;

exports[`SideBarItem renders SideBarItem with a label and icon without highlight prop 1`] = `
<MenuItem
  className="sidebar-item"
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        name="bell"
        size="large"
      />
    </span>
    <span>
      a label
    </span>
  </div>
</MenuItem>
`;
